assets/inject/semver-workflow/.github/workflows/main.yml

```
@@ -1,19 +1,32 @@
+---
 name: CI & Release
+
 on:
-  # Build on pushes to release branches
-  push:
-    branches: []
-  # Build on pull requests targeting release branches
+  # Build on pushes branches that have a PR (including drafts)
   pull_request:
+  # Build on commits pushed to branches without a PR if it's in the allowlist
+  push:
     branches: []
```

What's the reason for us not prefilling this:
```suggestion
    branches: [main]
```
Is this done for you during the script somewhere I've missed?

---

We have different setups throughout, but putting main as a start is probably the sane thing moving forward. Lets add it.

---

Where quite a few variations, but you are right, having main here as the default is the sane choice. Lets do it.